### ISSUE TITLE
Is this framework capable with multi-gpu simulation or managed by Ray?

### ISSUE THREAD

**Issue by Sephirex-X** [waymo-research/waymax#16]
Hi, thanks for this amazing effort.
I'm wondering if your guys can give some more examples about how to do RL under your framework by either multi-gpu simulation or Ray, if it's possible? 
thx a lot

**Comment by Sephirex-X** [comment]
BTW, is it possible to provide some examples about how to load WOMD locally, since all the configs provided are based on google cloud, many thx!

**Comment by justinjfu** [comment]
Hi @Sephirex-X,

Thanks for the question - we're gathering feedback like this and will add additional documentation based on what users are having questions about.

For now - you can do multi-GPU simulation by (a) passing `distributed=True` to the dataset function which will add an additional batch dimension to the data, and (b) using the `jax.pmap` decorator to distribute across multiple GPUs.

Basically, your code will have this structure:
```
data_config = dataclasses.replace(_config.WOD_1_1_0_TRAINING, distributed=True)
dataset = dataloader.simulator_state_generator(config=config)

@jax.pmap
def simulate(scenario: datatypes.SimulatorState):
   ... your code here ...

simulate(next(dataset))
```

Related issue: https://github.com/waymo-research/waymax/issues/9

**Comment by justinjfu** [comment]
For your other questions:

We haven't experimented with Ray yet. In my opinion a good path to integrate would be to write a wrapper to convert Waymax to Ray's environment API, similar to what we already have for DMEnv (https://github.com/waymo-research/waymax/tree/main/waymax/env/wrappers), but this is not on our timeline at the moment.

> BTW, is it possible to provide some examples about how to load WOMD locally, since all the configs provided are based on google cloud, many thx!

Tou can pass in a local path to the dataset config to use a local dataset (see https://github.com/waymo-research/waymax/blob/main/waymax/config.py#L39)

```
data_config = DatasetConfig(
    path='local_path/abc/xyz.tfrecord@1000',
    max_num_rg_points=20000,
    data_format=DataFormat.TFRECORD,
)
```

**Comment by Sephirex-X** [comment]
Thank you every much!

**Comment by Sephirex-X** [comment]
> Hi @Sephirex-X,
> 
> Thanks for the question - we're gathering feedback like this and will add additional documentation based on what users are having questions about.
> 
> For now - you can do multi-GPU simulation by (a) passing `distributed=True` to the dataset function which will add an additional batch dimension to the data, and (b) using the `jax.pmap` decorator to distribute across multiple GPUs.
> 
> Basically, your code will have this structure:
> 
> ```
> data_config = dataclasses.replace(_config.WOD_1_1_0_TRAINING, distributed=True)
> dataset = dataloader.simulator_state_generator(config=config)
> 
> @jax.pmap
> def simulate(scenario: datatypes.SimulatorState):
>    ... your code here ...
> 
> simulate(next(dataset))
> ```
> 
> Related issue: #9

hi when i run this(single GPU), i got huge mem consume but low efficiency, how can i improve the computation efficiency?
![image](https://github.com/waymo-research/waymax/assets/55309344/b07fda1a-6af1-42da-bbcf-9ea5f4b8c83f)
